drivers/counter/counter_mcux_lpc_rtc.c

```diff
@@ -132,15 +132,22 @@ static int mcux_lpc_rtc_set_top_value(const struct device *dev,
 	const struct mcux_lpc_rtc_config *config =
 			CONTAINER_OF(info, struct mcux_lpc_rtc_config, info);
 	struct mcux_lpc_rtc_data *data = dev->data;
+	uint32_t enable = (config->base->CTRL & RTC_CTRL_RTC_EN_MASK);
 
 	if (cfg->ticks != info->max_top_value) {
-		LOG_ERR("Wrap can only be set to 0x%x.", info->max_top_value);
+		LOG_DBG("Wrap can only be set to 0x%x.", info->max_top_value);
 		return -ENOTSUP;
 	}
 
-	if (!(cfg->flags & COUNTER_TOP_CFG_DONT_RESET)) {
+	if (enable) {
 		RTC_StopTimer(config->base);
+	}
+
+	if (!(cfg->flags & COUNTER_TOP_CFG_DONT_RESET)) {
 		RTC_SetSecondsTimerCount(config->base, 0);
+	}
+
+	if (enable) {
 		RTC_StartTimer(config->base);
```

Comment:
this function is deprecated, let's use `RTC_EnableTimer(base, enable)` and remove if statement

Reply:
Thanks for catching this. Update to include this change.